packages/api/src/db/setting.ts

```diff
@@ -0,0 +1,47 @@
+export type Setting<T = unknown> = {
+  id: string
+
+  name: string
+  value: T | null
+  settingRestriction?: SettingRestriction
+}
+
+export interface SettingInput<T> {
```

Comment:
maybe we can reuse SettingInput in create and update methods? similar as in e.g. subscriptions or payments

Reply:
For `create` not ,because I thought its best if the restrictions cannot be updated. Perhaps there's a more concise way to set this up though.

Reply:
We could do `export type SeetingInput<T = unknown> = Pick<Setting<T>,'value'>`.